__TESTS__/unit/actions/Variable.test.ts

```diff
@@ -0,0 +1,62 @@
+import Variable, {variable} from '../../../src/actions/variable/Variable';
+import TransformableImage from '../../../src/transformation/TransformableImage';
+import CloudinaryConfig from "../../../src/config/CloudinaryConfig";
+
+const CONFIG_INSTANCE = new CloudinaryConfig({
+  cloud: {
+    cloudName: 'demo'
+  }
+});
+
+describe('Tests for Transformation Action -- Rotate', () => {
+  it('Ensures new Rotate is accepted as an action to TransformableImage', () => {
+    const tImage = new TransformableImage();
+    // Ensures it compiles and doesn't throw
+    expect(
+      tImage.variable(new Variable('a', 30))
+    ).toEqual(tImage);
+  });
+  it('Ensures angle is accepted as an action to TransformableImage', () => {
+    const tImage = new TransformableImage();
+    // Ensures it compiles and doesn't throw
+    expect(
+      tImage.variable(variable('a', 30))
+    ).toEqual(tImage);
+  });
+  it('Creates a cloudinaryURL with variable', () => {
+    const url = new TransformableImage()
+      .setConfig(CONFIG_INSTANCE)
+      .variable(variable('a', 30))
+      .setPublicID('sample')
+      .toURL();
+
+    expect(url).toBe('http://res.cloudinary.com/demo/image/upload/$a_30/sample');
+  });
+  it('Creates a cloudinaryURL with variable', () => {
+    const url = new TransformableImage()
+      .setConfig(CONFIG_INSTANCE)
+      .variable(variable('a', 'bbb'))
```

Comment:
looks like a good start but there are some cases we are not taking care of. If this is something that we want to add to in the future then we should track the cases that we still did not implement. 
For example,
`.variable(variable('list', ['b', 'b', 'b'))` should yield $list_!b:b:b!

Maybe we could look at the spec and track what was implemented and what was not?

Reply:
Let's add these use cases as skipped( or empty) tests, so we know to go back to them.

Reply:
I've added a TODO in Variable. we'll add the tests when the spec is ready